sql/core/src/main/java/org/apache/spark/sql/execution/RecordBinaryComparator.java

```diff
@@ -22,6 +22,12 @@
 
 public final class RecordBinaryComparator extends RecordComparator {
 
+  boolean isLittlenEndian;
+
+  public RecordBinaryComparator(boolean isLittlenEndian) {
```

Comment:
No need for this. It's a fixed property of the entire VM / JVM. Just:
`private static final boolean LITTLE_ENDIAN = ByteOrder.nativeOrder.equals(ByteOrder.LITTLE_ENDIAN);`

Reply:
updated